apps/meteor/client/sidebar/header/UserDropdown.tsx

```diff
@@ -176,20 +180,55 @@ const UserDropdown = ({ user, onClose }: UserDropdownProps): ReactElement => {
 				<>
 					<Option.Divider />
 					{showAdmin && <Option icon={'customize'} label={t('Administration')} onClick={handleAdmin}></Option>}
-					{accountBoxItems.map((item, i) => {
+					{accountBoxItems
+						.filter((item) => !item.isAppButtonItem)
+						.map((item, i) => {
+							const action = (): void => {
+								if (item.href) {
+									FlowRouter.go(item.href);
+									onClose();
+								}
+								if (item.sideNav) {
+									SideNav.setFlex(item.sideNav);
+									SideNav.openFlex();
+									onClose();
+								}
+							};
+
+							return (
+								<Option
+									icon={item.icon as any}
+									label={t(item.name as any)}
+									onClick={item.href || item.sideNav ? action : undefined}
+									key={i}
+								></Option>
+							);
+						})}
+				</>
+			)}
+
+			{appBoxItems().length > 0 && (
+				<>
+					<Option.Divider />
+					<Box pi='x16' fontScale='c1' textTransform='uppercase'>
+						{t('Apps')}
+					</Box>
+					{appBoxItems().map((item, key) => {
 						const action = (): void => {
-							if (item.href) {
-								FlowRouter.go(item.href);
-								onClose();
-							}
-							if (item.sideNav) {
-								SideNav.setFlex(item.sideNav);
-								SideNav.openFlex();
-								onClose();
-							}
+							const { msg } = messageArgs(item.context);
+							triggerActionButtonAction({
+								rid: msg ? msg.rid : '',
+								mid: msg ? msg._id : '',
+								actionId: item.actionId,
+								appId: item.appId,
+								payload: { context: item.context },
+							});
 						};
-
-						return <Option icon={item.icon} label={t(item.name)} onClick={item.href || item.sideNav ? action : undefined} key={i}></Option>;
+						return (
+							<>
+								<Option label={t(item.name as any)} key={item.actionId + key} onClick={action as any} />
```

Comment:
Avoid using `any`. What problem did you face that made you go this way?

Reply:
in the case of the label prop, it requires to be a giant list of types, as you can see below:

`
'"view-logs" | "manage-emoji" | "manage-sounds" | "view-statistics" | "manage-oauth-apps" | "view-privileged-setting" | "manage-selected-settings" | "view-room-administration" | ... 5012 more ... | "onboarding.form.standaloneServerForm.manuallyIntegrate"'

so I decided to set `as any` to make it easy for this moment
`

Reply:
I think for `label` it makes sense, because translation strings coming from apps are dynamic and cannot be typed.

But for the `onClick` prop, is it possible to provide the proper typing?